packages/chat-headless/src/HeadlessProvider.ts

```diff
@@ -0,0 +1,14 @@
+import { ChatHeadless, ChatHeadlessImpl } from "./ChatHeadless";
+import { HeadlessConfig, ChatClient } from "./models";
+
+/**
+ * Provide an instance of {@link ChatHeadless} with all functionality built in.
+ *
+ * @public
+ */
+export function provideChatHeadless(
+  config: HeadlessConfig,
+  chatClient?: ChatClient
```

Comment:
during a discussion with Tom, I believe we both expressed concern for having support for two level of abstractions within one function/constructor (config that extends the core config AND a optional client to also create core). Having user pass in their own client also cause conflict with the core dependency version we have.

I think we should remove this second param. For public interface, user just need to pass in the config. We can then make a provideChatHeadlessInternal to accept the internalConfig and uses `provideChatCoreInternal`. This provide layer would create the ChatCore instance, and the ChatHeadless class's constructor will accept a ChatCore instance.

Reply:
Updated to match this proposal 👍